### PR TITLE
Refactor navigation to fix eslint type errors

### PR DESCRIPTION
**Pull request by anniehuo**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- revamped navigation to fix navigation type errors
- added types for navigation stack (mirrored vouchers for veggies implementation)
- added new stack navigator files for each flow
- removed old getauth functions from auth screens
- deleted old nav types screen
- moved fonts into fonts file 

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"

## How to review
[//]: # "Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"
- go through auth flow to make sure navigation still works
- use bottom tabs to navigate to make sure everything works
## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # "Optional - copy links to any tutorials or documentation that was useful to you when working on this PR"
vouchers for veggies navigation files: https://github.com/calblueprint/vouchers4veggies/tree/main/src/navigation

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @surajr10 @themiachou8

### CODE REVIEW

File: src/navigation/BottomTabNavigator.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import React, { useEffect } from 'react';
+// import { createNativeStackNavigator } from "@react-navigation/native-stack";
+// import { BottomNavigation } from '@material-ui/core';
```
Comment (wangannie) [inline src/navigation/BottomTabNavigator.tsx]: Remove comments!

File: src/navigation/BottomTabNavigator.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import { MaterialCommunityIcons } from '@expo/vector-icons';
+import { BottomTabParamList } from '../types/navigation';
+import FeedStackNavigator from './stacks/FeedStackNavigator';
+import DraftStackNavigator from './stacks/DraftStackNavigator';
+import ApprovalStackNavigator from './stacks/ApprovalStackNavigator';
+import SignoutStackNavigator from './stacks/SignOutStackNavigator';
+
+const Tab = createMaterialBottomTabNavigator<BottomTabParamList>();
+
+type Props = {
```
Comment (wangannie) [inline src/navigation/BottomTabNavigator.tsx]: Rename to `IconProps` to make it more descriptive

File: src/navigation/BottomTabNavigator.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  color: string;
+};
+
+const homeOutline = (props: Props) => {
```
Comment (wangannie) [inline src/navigation/BottomTabNavigator.tsx]: To make it cleaner, you can just do

`const ... = ({ color } : IconProps) => {`

which is destructuring the props!

Comment (wangannie) [inline src/navigation/BottomTabNavigator.tsx]: Would suggest renaming this to `HomeOutlineIcon` -- it should be PascalCase since it's a component

Comment (wangannie) [reply src/navigation/BottomTabNavigator.tsx]: Same idea for the other icons

**Review by wangannie — COMMENTED** [review]
Preliminary comments

**Review by surajr10 — APPROVED** [review]
